src/lib/crypto/pbkdf2.ts

```diff
@@ -0,0 +1,95 @@
+import { numberToBinUint32BE } from '../format/format.js';
+
+import type { HmacFunction} from './crypto.js';
+import { hmacSha256, hmacSha512 } from './crypto.js';
+
+export enum Pbkdf2Errors {
+  invalidIterationsError = 'Invalid PBKDF2 Parameters: Iterations must be a positive integer',
+  invalidDerivedKeyLengthError = 'Invalid PBKDF2 Parameters: Derived Key Length must be a positive integer',
+  invalidHmacLength = 'Invalid HMAC length: HMAC length must be a positive integer',
+}
+
+/**
+ * An object representing the parameters to use with PBKDF2 (Password-Based Key Derivation Function 2).
+ */
+export interface Pbkdf2Parameters {
+  /** The length of the derived key in bytes. */
+  derivedKeyLength: number;
+  password: Uint8Array;
+  iterations: number;
+  salt: Uint8Array;
+}
+
+export const instantiatePbkdf2Function =
+  (hmacFunction: HmacFunction, hmacByteLength: number) =>
+  (parameters: Pbkdf2Parameters) => {
+    /* eslint-disable functional/no-let, functional/no-loop-statement, functional/no-expression-statement, no-bitwise, no-plusplus */
+    const { password, salt, iterations, derivedKeyLength } = parameters;
+
+    if (!Number.isInteger(iterations) || iterations <= 0) {
```

Comment:
@bitjson If we include these checks, we break a linter "complexity" rule (max is 5, we end up on 10).

I'm not sure if these are maybe overkill? Or maybe should be split out into a separate `verifyPbkdf2Parameters` function?

Reply:
NOTE: If we keep these in, will try to add fail test cases for completeness.

Reply:
These checks are great! They make sense here, and I don't really see a reason to break them out into a separate function (it would be quite the stretch to use just this validation in some other context). So this is a good place to break our complexity rule (right above the relevant function):

```
// eslint-disable-next-line complexity
```

Reply:
Have placed this one in.